src/openms/source/FORMAT/DATAACCESS/SiriusMzTabWriter.cpp

```diff
@@ -45,7 +45,8 @@ using namespace std;
 
 int SiriusMzTabWriter::extract_scan_index(const String &path)
 {
-  return (path.substr(path.find_last_not_of("0123456789") + 1)).toInt();
+  boost::regex regexp_ind("--(?<SCAN>\\d+)--");
```

Comment:
does this work for Thermo and other files?

Reply:
In this case it is completly independent of any vendor, since it is used to parse a directory path via a regular expression. 

`1_20200921_144720_Olivers-MBP.fritz.box_58922_2_4024150408384032977-148005--779--Phenylpyruvica`

`<SCAN>` is only the group name which is used in the  `SpectrumLookup::extractScanNumber` method.